components/org.wso2.carbon.identity.application.authenticator.fido2.endpoint/src/main/java/org/wso2/carbon/identity/application/authenticator/fido2/endpoint/impl/StartRegistrationApiServiceImpl.java

```diff
@@ -52,6 +54,9 @@ public Response startRegistrationPost(String appId) {
         try {
             if (appId.contains(FIDO2Constants.EQUAL_OPERATOR)) {
                 appId = URLDecoder.decode(appId.split(FIDO2Constants.EQUAL_OPERATOR)[1], IdentityCoreConstants.UTF_8);
+            } else if (appId.startsWith("{")) {
```

Comment:
If possible, let's check the string is a valid JSON string, instead of checking for the starting parenthesis

Reply:
1. Need to check for a valid JSON, then parse and get the app ID.
2. Need to handle the error scenarios -> for invalid JSON, or if app id is not present etc.

Reply:
Fixed